ui/app/components/tx-list.js

```diff
@@ -0,0 +1,141 @@
+const Component = require('react').Component
+const connect = require('react-redux').connect
+const h = require('react-hyperscript')
+const inherits = require('util').inherits
+
+const valuesFor = require('../util').valuesFor
+
+module.exports = connect(mapStateToProps)(TxList)
+
+function mapStateToProps(state) {
+  return {
+    network: state.metamask.network,
+    unapprovedMsgs: valuesFor(state.metamask.unapprovedMsgs),
+    shapeShiftTxList: state.metamask.shapeShiftTxList,
+    transactions: state.metamask.selectedAddressTxList || [],
+    conversionRate: state.metamask.conversionRate,
+  }
+}
+
+inherits(TxList, Component)
+function TxList () {
+  Component.call(this)
+}
+
+const contentDivider = h('div', {
+  style: {
+    marginLeft: '1.3em',
+    marginRight: '1.3em',
+    height:'1px',
+    background:'#E7E7E7', // TODO: make custom color
+  },
+})
+
+TxList.prototype.render = function () {
+
+  const { transactions, network, unapprovedMsgs, conversionRate } = this.props
+
+  var shapeShiftTxList
+  if (network === '1') {
+    shapeShiftTxList = this.props.shapeShiftTxList
+  }
+  const txsToRender = !shapeShiftTxList ? transactions.concat(unapprovedMsgs) : transactions.concat(unapprovedMsgs, shapeShiftTxList)
+  .sort((a, b) => b.time - a.time)
+
+  console.log("transactions to render", txsToRender)
+
+  return h('div.flex-column.tx-list-container', {}, [
```

Comment:
Right, that's something I would need some guidance on down the road - the current `tx-view` has a relatively flat DOM structure and when I am extracting components like `balance-component` and `tx-list`, I always need to wrap an extra div to make sure it works. I am new to flex layouts, and I seemed to get lucky last time with `balance-component` and didn't mess up too much. Would love to maybe pair with you on fixing this.